src/components/SearchResults/SearchError.tsx

```diff
@@ -0,0 +1,93 @@
+import { Heading, Flex, Link, Text } from "@nypl/design-system-react-components"
+import type { HTTPStatusCode } from "../../types/appTypes"
+import { appConfig } from "../../config/config"
+import { SITE_NAME } from "../../config/constants"
+import RCHead from "../Head/RCHead"
+import Layout from "../Layout/Layout"
+import ExternalLink from "../Links/ExternalLink/ExternalLink"
+import { useContext } from "react"
+import { FeedbackContext } from "../../context/FeedbackContext"
+
+type SearchErrorProps = {
+  errorStatus: HTTPStatusCode
+}
+
+export default function SearchError({ errorStatus }: SearchErrorProps) {
+  const metadataTitle = `${errorStatus} | ${SITE_NAME}`
+  const { onOpen } = useContext(FeedbackContext)
+  const errorContent = (status) => {
+    switch (status) {
+      case 404:
+        return (
+          <>
+            <Heading level="h3" tabIndex={-1} id="search-results-heading">
+              No results found.
+            </Heading>
+            <p>
+              Try a different search, or search our{" "}
+              <ExternalLink href={appConfig.urls.legacyCatalog}>
+                Legacy Catalog
+              </ExternalLink>{" "}
+              for research materials.
+            </p>
+          </>
+        )
+      case 500:
+        return (
+          <>
+            <Heading level="h3" tabIndex={-1} id="search-results-heading">
+              Something went wrong on our end.
+            </Heading>
+            <Text marginBottom="0">
+              We encountered an error while trying to load the page.
+            </Text>
+            <Text marginBottom="0">
+              Try refreshing the page or{" "}
+              <Link onClick={onOpen} id={"feedback-link"}>
+                contact us
+              </Link>{" "}
+              if the error persists.
+            </Text>
+          </>
+        )
+      default:
+        // 400, 422, or other generic client errors
+        return (
+          <>
+            <Heading level="h3" tabIndex={-1} id="search-results-heading">
+              There was an unexpected error.
+            </Heading>
+            <Text marginBottom="0">
+              We couldn&apos;t process your request at this time.
+            </Text>
+            <Text marginBottom="0">
+              Try again later or{" "}
+              <Link onClick={onOpen} id={"feedback-link"}>
+                contact us
+              </Link>{" "}
+              if the error persists.
+            </Text>
+          </>
+        )
+    }
+  }
+  return (
+    <>
+      <RCHead metadataTitle={metadataTitle} />
+      <Layout activePage={"search"}>
+        <Flex
+          flexDir="column"
+          marginTop="l"
+          marginBottom="xxl"
+          marginLeft="l"
+          marginRight="l"
+          alignItems="center"
+          justifyContent="center"
+          textAlign="center"
+        >
+          {errorContent(errorStatus)}
```

Comment:
Instead of an `errorContent` function that needs to be created earlier in the component and then called later, you could create a variable `let errorContent` and then add your switch statement so that the value is already determined by the time it gets to the return statement. This is not a blocker, and likely does not make a big difference in this component, but is generally a better practice to reduce unnecessary work.

```
  let errorContent

  switch (errorStatus) {
    case 404:
      errorContent = (
```